src/process/plumerise/CCPr_Plumerise_Mod.F90

```diff
@@ -0,0 +1,271 @@
+                           MetState%V,                                  &
+                           MetState%PMID,                               &
+                           MetState%HFLUX,                              &
+                           MetState%PBLH,                               &
+                           MetState%USTAR,                              &
+                           MetState%T2M,                                &
+                           MetState%PS,                                 &
+                           SIZE(MetState%T),                            &
+                           EmisState%Cats(c)%Species(s)%STKDM(p),   &
+                           EmisState%Cats(c)%Species(s)%STKHT(p),   &
+                           EmisState%Cats(c)%Species(s)%STKTK(p),   &
+                           EmisState%Cats(c)%Species(s)%STKVE(p),   &
+                           plmHGT,                                      &
+                           EFRAC)
+                        if (RC /= CC_SUCCESS) then
+                           errMsg = 'Error in CCPr_Briggs_Plumerise'
+                           CALL CC_Error( errMsg, RC, thisLoc )
+                        endif
+
+                        do z = 1, GridState%number_of_levels
+                           ColEmis(z) = ColEmis(z) + EmisState%Cats(c)%Species(s)%PlmSrcFlx(p) * EFRAC(z)
+                        enddo
+
+                     end if ! plume source loop
+
+                     if (verbose .eqv. .true.) then
+                        write(*,*) '------------ PLUMERISE ---------------'
+                        write(*,*) 'Category: ', EmisState%Cats(c)%Name
+                        write(*,*) 'Species: ', EmisState%Cats(c)%Species(s)%Name
+                        write(*,*) 'PlmHGT: ', plmHGT
+                        write(*,*) 'PlmSrcFlx: ', EmisState%Cats(c)%Species(s)%PlmSrcFlx(p)
+                        if (EmisState%Cats(c)%Species(s)%plumerise == 2) then
+                           write(*,*) 'PlumriseOpt: BRIGGS'
+                           write(*,*) '  STKDM: ', EmisState%Cats(c)%Species(s)%STKDM(p)
+                           write(*,*) '  STKHT: ', EmisState%Cats(c)%Species(s)%STKHT(p)
+                           write(*,*) '  STKTK: ', EmisState%Cats(c)%Species(s)%STKTK(p)
+                           write(*,*) '  STKVE: ', EmisState%Cats(c)%Species(s)%STKVE(p)
+                        else if (EmisState%Cats(c)%Species(s)%plumerise == 1) then
+                           write(*,*) 'PlumriseOpt: SOFIEV'
+                           write(*,*) '  FRP: ', EmisState%Cats(c)%Species(s)%frp(p)
+                        endif
+                     endif
+                  end do ! p - Plumerise Source Loop
+
+                  ! Add emission to ColEmis to Species total flux in grid cell
+                  EmisState%Cats(c)%Species(s)%Flux = ColEmis
+
+               enddo ! s - Species List Loop
```

Comment:
Just noting could use named `do` instead of these comments, like
```
species: do i = 1, n
   ...
end do species
```

Reply:
ooo tell me more

Reply:
```f90
species: do i = 1, n
   ! do some stuff
end do species
```
That's the syntax, you could give it a go.